src/sentry/logging/handlers.py

```diff
@@ -83,3 +86,40 @@ def emit(self, record, logger=get_logger()):
                 kwargs['positional_args'] = (record.args, )
 
         logger.log(**kwargs)
+
+
+class MessageContainsFilter(logging.Filter):
+    """
+    A logging filter that allows log records where the message
+    contains given substring(s).
+
+    contains -- a string or list of strings to match
+    """
+
+    def __init__(self, contains):
+        if not isinstance(contains, list):
+            contains = [contains]
+        if not all(isinstance(c, six.string_types) for c in contains):
+            raise TypeError("'contains' must be a string or list of strings")
+        self.contains = contains
+
+    def filter(self, record):
+        message = record.getMessage()
+        return any(c in message for c in self.contains)
+
+
+class MetricsLogHandler(logging.Handler):
+    def emit(self, record, logger=get_logger()):
+        """
+        Turn something like:
+            > django.request: Forbidden (CSRF cookie not set.): /account
```

Comment:
Is `django.requst` the `record.name` below?

So is it really converting: `django.request.Forbidden (CSRF cookie not set.): /account`?

I only ask because I tried to run this on that string to verify and got: `django.request_forbidden_csrf_cookie_not_set._account`

Reply:
Oops, yeah that comment is wrong, the `record.name` is the logger name, so yeah `django.request`. And I'm putting a dot between them. Definitely also open to other suggestions for how to turn django log messages into sane metric names.